tests fail on macos locally, but not on CI

specifically, there is an issue with the `distance`   function when using `node_attr_dtypes = {"position": "int[4]"},` (but *not* `double[4]`), and it only seems to happen when I build it locally.  So there must be some differences in linker behavior with whatever build chain  I have vs what github has:

```sh
➜ pytest tests/test_spatial_graph.py::test_construction -v -s -x
============================================================================================================ test session starts =============================================================================================================
platform darwin -- Python 3.12.10, pytest-8.3.5, pluggy-1.5.0 -- /Users/talley/dev/self/spatial_graph/.venv/bin/python3
cachedir: .pytest_cache
rootdir: /Users/talley/dev/self/spatial_graph
configfile: pyproject.toml
collected 48 items

tests/test_spatial_graph.py::test_construction[True-edge_attr_dtypes0-node_attr_dtypes0-int8] PASSED
tests/test_spatial_graph.py::test_construction[True-edge_attr_dtypes0-node_attr_dtypes0-uint8] PASSED
tests/test_spatial_graph.py::test_construction[True-edge_attr_dtypes0-node_attr_dtypes0-int16] PASSED
tests/test_spatial_graph.py::test_construction[True-edge_attr_dtypes0-node_attr_dtypes0-uint16] PASSED
tests/test_spatial_graph.py::test_construction[True-edge_attr_dtypes0-node_attr_dtypes1-int8] FAILED

================================================================================================================== FAILURES ==================================================================================================================
______________________________________________________________________________________ test_construction[True-edge_attr_dtypes0-node_attr_dtypes1-int8] ______________________________________________________________________________________

node_dtype = 'int8', node_attr_dtypes = {'position': 'int[4]'}, edge_attr_dtypes = {}, directed = True

    @pytest.mark.parametrize("node_dtype", node_dtypes)
    @pytest.mark.parametrize("node_attr_dtypes", node_attr_dtypes)
    @pytest.mark.parametrize("edge_attr_dtypes", edge_attr_dtypes)
    @pytest.mark.parametrize("directed", [True, False])
    def test_construction(node_dtype, node_attr_dtypes, edge_attr_dtypes, directed):
>       sg.SpatialGraph(
            ndims=4,
            node_dtype=node_dtype,
            node_attr_dtypes=node_attr_dtypes,
            edge_attr_dtypes=edge_attr_dtypes,
            position_attr="position",
            directed=directed,
        )

tests/test_spatial_graph.py:22:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
spatial_graph/spatial_graph.py:29: in __init__
    self._edge_rtree = LineRTree(f"{node_dtype}[2]", self.coord_dtype, ndims)
spatial_graph/rtree/rtree.py:100: in __new__
    wrapper = witty.compile_module(
../witty/src/witty/compile_module.py:118: in compile_module
    return _load_dynamic(module_name, module_lib)
../witty/src/witty/compile_module.py:151: in _load_dynamic
    module = importlib.util.module_from_spec(spec)
<frozen importlib._bootstrap>:813: in module_from_spec
    ???
<frozen importlib._bootstrap_external>:1293: in create_module
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

f = <built-in function create_dynamic>
args = (ModuleSpec(name='_witty_module_aa34c19376cf86f5fd5ec2461fa68dfb', loader=<_frozen_importlib_external.ExtensionFileLoa...gin='/Users/talley/Library/Caches/cython/witty/_witty_module_aa34c19376cf86f5fd5ec2461fa68dfb.cpython-312-darwin.so'),)
kwds = {}

>   ???
E   ImportError: dlopen(/Users/talley/Library/Caches/cython/witty/_witty_module_aa34c19376cf86f5fd5ec2461fa68dfb.cpython-312-darwin.so, 0x0002): symbol not found in flat namespace '_distance'

<frozen importlib._bootstrap>:488: ImportError
============================================================================================================== warnings summary ==============================================================================================================
.venv/lib/python3.12/site-packages/Cheetah/Template.py:25
  /Users/talley/dev/self/spatial_graph/.venv/lib/python3.12/site-packages/Cheetah/Template.py:25: DeprecationWarning: 'cgi' is deprecated and slated for removal in Python 3.13
    import cgi  # Used by .webInput() if the template is a CGI script.

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================================================================== short test summary info ===========================================================================================================
FAILED tests/test_spatial_graph.py::test_construction[True-edge_attr_dtypes0-node_attr_dtypes1-int8] - ImportError: dlopen(/Users/talley/Library/Caches/cython/witty/_witty_module_aa34c19376cf86f5fd5ec2461fa68dfb.cpython-312-darwin.so, 0x0002): symbol not found in flat namespace '_distance'
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,3 +1,7 @@
+[build-system]
+requires = ["hatchling"]
+build-backend = "hatchling.build"
+
 [project]
 name = "spatial-graph"
 version = "0.0.1"
@@ -5,28 +9,19 @@ description = "A spatial graph datastructure for python."
 readme = "README.md"
 requires-python = ">=3.9"
 license = { text = "MIT" }
-authors = [
-    { email = "funkej@janelia.hhmi.org", name = "Jan Funke" },
-]
-dependencies = [
-  "witty>=v0.2.0",
-  "cheetah3",
-  "numpy",
-  "setuptools>=75.8.0",
-]
+authors = [{ email = "funkej@janelia.hhmi.org", name = "Jan Funke" }]
+dependencies = ["witty>=v0.2.0", "CT3", "numpy", "setuptools>=75.8.0"]
 
 [project.urls]
 homepage = "https://github.com/funkelab/spatial_graph"
 repository = "https://github.com/funkelab/spatial_graph"
 
 [project.optional-dependencies]
-dev = [
-    "pytest>=8.3.4",
-]
+dev = ["pytest>=8.3.4"]
 
 [tool.hatch.metadata]
 allow-direct-references = true
 
-[build-system]
-requires = ["hatchling"]
-build-backend = "hatchling.build"
+[tool.pytest.ini_options]
+testpaths = ["tests"]
+filterwarnings = ["ignore:'cgi' is deprecated:DeprecationWarning"]

--- a/spatial_graph/graph/graph.py
+++ b/spatial_graph/graph/graph.py
@@ -8,10 +8,15 @@ from ..dtypes import DType
 # Set platform-specific compile arguments
 if sys.platform == "win32":
     # Use /O2 for optimization and /std:c++20 for C++20
-    EXTRA_COMPILE_ARGS = ["/O2", "/std:c++20"]
+    EXTRA_COMPILE_ARGS = ["/O2", "/std:c++20", "/wd4101"]
 else:
     # -O3 for optimization and -std=c++20 for C++20
-    EXTRA_COMPILE_ARGS = ["-O3", "-std=c++20"]
+    EXTRA_COMPILE_ARGS = [
+        "-O3",
+        "-std=c++20",
+        "-Wno-unused-variable",
+        "-Wno-unreachable-code",
+    ]
 
 
 class Graph:

--- a/spatial_graph/rtree/line_rtree.py
+++ b/spatial_graph/rtree/line_rtree.py
@@ -91,7 +91,7 @@ inline coord_t point_segment_dist2(const coord_t point[], const coord_t start[],
     return length2(a);
 }
 
-inline coord_t distance(const coord_t point[], const struct rect *rect, const struct item_t item) {
+extern inline coord_t distance(const coord_t point[], const struct rect *rect, const struct item_t item) {
     coord_t start[DIMS];
     coord_t end[DIMS];
     for (int d = 0; d < DIMS; d++) {

--- a/spatial_graph/rtree/rtree.py
+++ b/spatial_graph/rtree/rtree.py
@@ -7,6 +7,10 @@ from pathlib import Path
 from ..dtypes import DType
 
 DEFINE_MACROS = [("RTREE_NOATOMICS", "1")] if sys.platform == "win32" else []
+if sys.platform == "win32":
+    EXTRA_COMPILE_ARGS = ["/O2"]
+else:
+    EXTRA_COMPILE_ARGS = ["-O3", "-Wno-unreachable-code"]
 
 
 class RTree:
@@ -100,7 +104,7 @@ class RTree:
                 src_dir / "src" / "rtree.c",
                 src_dir / "src" / "config.h",
             ],
-            extra_compile_args=["/O2" if sys.platform == "win32" else "-O3"],
+            extra_compile_args=EXTRA_COMPILE_ARGS,
             include_dirs=[str(src_dir)],
             language="c",
             quiet=True,

